dlfcn_android_64bit.go

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2024 The Ebitengine Authors
+
+//go:build android && (amd64 || arm64)
```

Comment:
I slightly prefer `android && !386 && !arm`

Optional: another option is integrate `dlfnc_android_32bit.go` and `dlfnc_android_64bit.go` and use a little complex calculation like https://cs.opensource.google/go/go/+/master:src/runtime/malloc.go;l=210:

```go
const (
    _64bit = 1 << (^uintptr(0) >> 63) / 2
    RTLD_DEFAULT = (1 - _64bit) * 0xffffffff
    RTLD_LAZY = 0x00000001
    RTLD_NOW = _64bit * 0x00002
    RTLD_LOCAL = 0x00000000
    RTLD_GLOBAL = _64bit * 0x00100 + (1 - _64bit) * 0x00000002
)
```